### PR TITLE
Scripts/Broken Isles: Warrior Order Hall - some NPC spawns and scripts

### PR DESCRIPTION
**Pull request by Golrag**
**Changes proposed:**

- Add some spawns for warrior order hall & waypoints (those who are spawned before you turn in the quest [Odyn and the Valarjar](https://www.wowhead.com/quest=39654/odyn-and-the-valarjar))
- Some scripts for npc's
- Script for introduction scene with Odyn
- Script for Danica
- Added a graveyard

**Target branch(es):** master
-  master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:**
- Tested in game
    - Looked at every NPC's behaviour to check if they behaved as I intended to.
- Builds

**Known issues and TODO list:** (add/remove lines as needed)
- DB Waypoints aren't smooth
- ~~Val'kyrs are too slow.~~
- Timers may not be 100% blizzlike.
- Danica the reclaimer should personally despawn ~~and a personal one should be escorting you.~~
- ~~Infinite Scene due to quest objective order bug (no areatrigger spotted so I made the trigger in MoveInLoS).~~
- Val'kyrs jump is weird.
- Missing emotes for npc's using waypoints.
- ~~Phase condition not working~~
- Some creatures respawn at their corpse instead of their spawn position.
- Sometimes a val'kyr spawns in the ground.

Any feedback & help is welcome. I feel like some scripts could have been easier.

### CODE REVIEW

File: sql/updates/world/master/2017_mm_dd_xx_world.sql [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+(@CGUID+116, 0, 0, 0, 1, 0, 0, 0, 0, ''); -- Valarjar Aspirant
+
+DELETE FROM `creature_template_addon` WHERE `entry` IN (93823);
+INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `aiAnimKit`, `movementAnimKit`, `meleeAnimKit`, `auras`) VALUES
+(93823, 0, 0, 50331648, 1, 0, 0, 0, 0, ''); -- Danica the Reclaimer
+
+-- Pathing for Quartermaster Durnolf
+SET @PATH := (@CGUID+0) * 10;
+
+DELETE FROM `creature_addon` WHERE `guid`=@CGUID+16;
```
Comment (Golrag) [inline sql/updates/world/master/2017_mm_dd_xx_world.sql]: should be +0

File: src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp [hunk 2]
```diff
@@ -0,0 +1,885 @@
...
+
+        void MovementInform(uint32 /*type*/, uint32 id) override
+        {
+            switch (id)
+            {
+                case 0:
+                    _scheduler.Schedule(Seconds(10), [this](TaskContext /*context*/)
+                    {
+                        Player* player = ObjectAccessor::FindConnectedPlayer(_summonerGuid);
+                        me->GetMotionMaster()->MoveSmoothPath(1, DanicaPath02, DanicaPath02Size, false, true);
```
Comment (Shauren) [inline src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp]: Are you sure about all of your uses of MoveSmoothPath? Is Catmullrom spline flag present in sniff? If not then don't use this function

Comment (Golrag) [inline src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp]: No it's not, shouldn't the function only set catmullroll for flying creatures only and smoothgroundpath for non-flying units ?
Which function should I use if I want an NPC to move by path ?
Small exception for danica, she is not using catmullroll while flying, but I can't find a function for it except ```me->GetMotionMaster()->MovePath()``` which requires the path to be in the database.

**Comment by Golrag** [comment]
>Infinite Scene due to quest objective order bug (no areatrigger spotted so I made the trigger in MoveInLoS).

Fixed the order but I still have the infinite scene, will look at it tomorrow or sometime later.

**Comment by Golrag** [comment]
Noticed a bug, when the val'kyrs reached their last point and DisappearAndDie() is used that's the spot where they will respawn (they shouldn't respawn there).

**Comment by Golrag** [comment]
Added some more issues to the known issues section of this PR. Some of them I can not fix (in this PR), some I'm not sure about (looking at the move jump of the val'kyrs). And I don't want to go to my sniffs again to find which emotes I need to add to the npc's using db waypoints in the mead hall.

Anything left I need to do for this PR ? :p

**Comment by Golrag** [comment]
I could use some help to understanding what's wrong with the val'kyrs.. I tried checking their position on Reset, it was good but they did not respawn at that position tho, and with the latest changes they "move down" above the hole and despawn instantly :/
Small note to this: I'm using max visibilty distance for continents (533) to have a better view of the whole order hall (it looks nicer)

**Comment by Golrag** [comment]
>Sometimes a valkyr spawns in the ground.

https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/Creature/Creature.cpp#L1433-L1442
Don't know why it's there so i'm not touching it, but pretty sure it's the cause :p

**Comment by Golrag** [comment]
@Shauren anything I will have to change here for your phasing PR ?
Also a bit more off-topic but is there a better way of finding out in which phase a creature/gameobject is in when your smsg_phase_shift lists a lot of phases? As it was really a mess to find these

**Comment by Shauren** [comment]
Phasing rewrite should not break any existing uses, only introduce new functionality (and fix areatriggers)
As for finding phase id... there really isnt a better way

**Comment by Aokromes** [comment]
@Golrag https://app.codacy.com/app/TrinityCore/TrinityCore/pullRequest?prid=1497762

**Comment by Golrag** [comment]
@Aokromes fixed

File: src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp [hunk 3]
```diff
@@ -0,0 +1,922 @@
...
+    ITEM_137176                         = 137176, // No name found on wowhead.
+    ITEM_137263                         = 137263, // No name found on wowhead.
+    ITEM_137265                         = 137265 // No name found on wowhead.
+};
+
+struct npc_danica_the_reclaimer : public ScriptedAI
+{
+    npc_danica_the_reclaimer(Creature* creature) : ScriptedAI(creature) { }
+
+    static uint8 const DanicaPath01Size = 6;
```
Comment (Shauren) [inline src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp]: Do the std::extent thing

**Review by Shauren — CHANGES_REQUESTED** [review]

**Comment by Golrag** [comment]
@Shauren done
Also I noticed that the first text of the personal spawn could be seen by an other character

**Comment by Stanwarr** [comment]
Any idea why this isn't on the master branch anymore?

**Comment by illfated** [comment]
What do you mean? This PR has not been merged yet, so I guess you are referring to something else. 😕 ❔

**Comment by Golrag** [comment]
Updated to https://github.com/TrinityCore/TrinityCore/commit/2704e3c6f8aba47ed3d753ef06aa3c5b9c196a57

Anything left for me to do with this PR?

**Comment by Golrag** [comment]
Updated to 6b29c4da76fbdcb55b4a89f485a5cff58c4296b9

Haven't done any tests as I get an error when I try to connect (BLZ51901017). Once that's fixed I will test

**Comment by Golrag** [comment]
Seems like everything still works like I last tested it

**Comment by illfated** [comment]
Maybe we could have a comment from Shauren whether he thinks this is OK.

**Pull request merged**